pkg/storage/bloom/v1/reader.go

```diff
@@ -113,17 +122,28 @@ func (r *DirectoryBlockReader) Blooms() (io.ReadSeeker, error) {
 }
 
 func (r *DirectoryBlockReader) TarEntries() (iter.Iterator[TarEntry], error) {
+	var err error
 	if !r.initialized {
-		if err := r.Init(); err != nil {
+		if err = r.Init(); err != nil {
 			return nil, err
 		}
 	}
 
+	_, err = r.index.Seek(0, io.SeekStart)
```

Comment:
Q curious: why are we adding this? Was there a bug?

Reply:
This wasn't used at the write path before, so we never had the issue. Without seeking, the position is at the end of the file, leading to an empty file when uploaded. This was discovered by your integration test.